[xenial] Nightly CI runs against Xenial are failing

## Description

The nightly CI run against Xenial is currently failing. Let's investigate and get it passing.

## Steps to Reproduce 

1. Inspect output from recent nightly CI run against Xenial, e.g. https://circleci.com/gh/freedomofpress/securedrop/22238
2. Observe failure on bouncing tor service

## Expected Behavior

Nightly CI run against Xenial passes. At the very least, the install flow succeeds, and tests report failure.

## Actual Behavior

Nightly CI run against Xenial fails, during the install step, and config tests are never run.

## Comments

Failure output from https://circleci.com/gh/freedomofpress/securedrop/22238:

```
 RUNNING HANDLER [tor-hidden-services : Register host name to wait for.] ********
    ok: [mon-staging]
    ok: [app-staging]
    
    RUNNING HANDLER [tor-hidden-services : restart tor (simple)] *******************
    fatal: [app-staging]: FAILED! => {"changed": false, "msg": " * Stopping tor daemon...\n   ...done.\n * Starting tor daemon...\n   ...fail!\n"}
    changed: [mon-staging]
    
    RUNNING HANDLER [tor-hidden-services : restart tor (async)] ********************
    
    RUNNING HANDLER [tor-hidden-services : Clear out SSH sessions, to prevent reuse of stale ControlPersist file.] ***
    
    RUNNING HANDLER [tor-hidden-services : Waiting for SSH connection (slow)...] ***
    ok: [mon-staging -> localhost]
    
    NO MORE HOSTS LEFT *************************************************************
```

We should confirm this error occurs reliably, by bouncing the workflow in Circle CI and confirming a second failure at the same step.

Got a different error on https://circleci.com/gh/freedomofpress/securedrop/22268:

```
#!/bin/bash -eo pipefail
make ci-go

./devops/gce-nested/ci-go.sh
Created [https://www.googleapis.com/compute/v1/projects/securedrop-ci/zones/us-west1-c/instances/sd-ci-nested-22268].
NAME                ZONE        MACHINE_TYPE  PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP   STATUS
sd-ci-nested-22268  us-west1-c  n1-highcpu-4               172.18.0.6   35.230.112.0  RUNNING
ssh: connect to host 35.230.112.0 port 22: Connection refused

rsync: connection unexpectedly closed (0 bytes received so far) [sender]
rsync error: unexplained error (code 255) at io.c(226) [sender=3.1.0]
make: *** [ci-go] Error 255
Exited with code 2
```

We may want to raise timeout or retry values. Rerunning as https://circleci.com/workflow-run/c06b29aa-3d3a-44c0-8607-17666a2ae6e3

So the tor failure to restart is an intermittent problem (potentially due to network instability?) that has happened a couple of times, I know @heartsucker saw that earlier this week and it was resolved via a rerun. 

The last instance of the nightly xenial job running was https://circleci.com/gh/freedomofpress/securedrop/22624, where the provisioning completed without error and all tests ran but the failure occurred due to:

```
scp: /home/sdci/securedrop-source/junit/*xml: No such file or directory
make: *** [ci-go-xenial] Error 1
Exited with code 2
```

See #4148 which resolves and gets the xenial CI job passing, we should file followups for other flakes/failures as they are discovered

--- a/molecule/libvirt-staging-xenial/molecule.yml
+++ b/molecule/libvirt-staging-xenial/molecule.yml
@@ -66,3 +66,4 @@ verifier:
   options:
     n: auto
     v: 2
+    junit-xml: ../../junit/testinfra-results.xml

